types/shares.go

```diff
@@ -99,7 +99,9 @@ func split(rawData []byte, nid namespace.ID) []NamespacedShare {
 	rawData = rawData[MsgShareSize:]
 	for len(rawData) > 0 {
 		shareSizeOrLen := min(MsgShareSize, len(rawData))
-		rawShare := []byte(append(nid, rawData[:shareSizeOrLen]...))
+		rawShare := make([]byte, NamespaceSize)
+		copy(rawShare, nid)
+		rawShare = append(rawShare, rawData[:shareSizeOrLen]...)
```

Comment:
So this is essentially the same logic (at rather the intended one) as before now but we use `copy` to make sure that the raw share contains the `nid` instead of using append 👍🏼

Reply:
yep :slightly_smiling_face:

Reply:
BTW, I guess the append version of this should have looked like this then:
```go
append(append(
	make([]byte, NamespaceSize),
	nid...),
	rawData[:shareSizeOrLen]...)
	
```